numpy/_core/tests/test_multiarray.py

```diff
@@ -5577,15 +5577,14 @@ def _create_data(self):
         x[0, :, 1] = [np.nan, np.inf, -np.inf, np.nan]
         return x
 
-    @pytest.fixture(params=["string", "path_obj"])
-    def tmp_filename(self, tmp_path, request):
+    def _create_filename(self, path, param):
         # This fixture covers two cases:
         # one where the filename is a string and
         # another where it is a pathlib object
-        filename = tmp_path / "file"
```

Comment:
Since the tests only depend on the file created, this could be much simpler if we used the same directory from all threads and then created one file for each thread, by appending a suffix with the thread id. The way I see things, this would be enough isolation for everything to be okay with less code.

Reply:
@lysnikolaou Unfortunately `thread_index` suffers from the same issue as `tmp_path` where it only gets updated when directly requested by a test. We could potentially try fixing this in `pytest-run-parallel`, but there's also the issue that fixtures generally get called only once per-test and don't have a "thread" scope. Hmmmm if we could get the thread id without calling into fixtures that would get rid of a lot of code here.